From oliver@neukum.org  Tue Jan 22 14:11:16 2008
From: Oliver Neukum <oliver@neukum.org>
Date: Tue, 22 Jan 2008 14:24:56 +0100
Subject: USB: fix memleak in ark3116 serial driver
To: ark3116_driver@auctionant.de, Greg KH <greg@kroah.com>, linux-usb@vger.kernel.org
Message-ID: <200801221424.57108.oliver@neukum.org>
Content-Disposition: inline


in an error case memory already allocated must be freed again.

Signed-off-by: Oliver Neukum <oneukum@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/serial/ark3116.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/usb/serial/ark3116.c
+++ b/drivers/usb/serial/ark3116.c
@@ -151,8 +151,10 @@ static int ark3116_attach(struct usb_ser
 	return 0;
 
 cleanup:
-	for (--i; i >= 0; --i)
+	for (--i; i >= 0; --i) {
+		kfree(usb_get_serial_port_data(serial->port[i]));
 		usb_set_serial_port_data(serial->port[i], NULL);
+	}
 	return -ENOMEM;
 }
 
